include/lk/snappy_boot_v2.h

```diff
@@ -209,6 +214,19 @@ typedef struct SNAP_RECOVERY_BOOT_SELECTION {
      */
     char recovery_system_status[SNAP_NAME_MAX_LEN];
 
+    /** device_lock_state contains the lock state of the device. It is used by the
+     * bootloader to track device lock changes. When lock state changes, device goes
+     * automatically to install mode. This entry is completely transparent
+     * to the snapd and is only modified by bootloader.
+     * Only last char in the aray is used (device_lock_state[0])
+     * Permitted values:
+     *  0: DEVICE_STATE_UNKNOW:   initial value at first boot.
+     *          This is changed by the bootloader to reflect actual device state.
+     *  1: DEVICE_STATE_UNLOCKED: unlocked device
+     *  2: DEVICE_STATE_LOCKED:   locked device
+     */
+    char device_lock_state[SNAP_NAME_MAX_LEN];
```

Comment:
I wonder if this shouldn't be more elaborate:
```
union {
    char as_bytes[SNAP_NAME_MAX_LEN];
    struct {
        char device_lock_state;
        char __pad[SNAP_NAME_MAX_LEN-1];
    };
} flags __attribute__((packed));
```

Reply:
This is not needed, as the actual handling of the cases is entirely down to the bootloader implementation details and can differ from bootloader to bootloader.
Idea of this PR is to ensure this variable slot is not repurposed by snapd in the future, colliding with bootloader use of the same variable slot.